lib/galaxy/tools/split_collection_by_number.xml

```diff
@@ -0,0 +1,101 @@
+<tool id="__SPLIT_COLLECTION_BY_NUMBER__"
+      name="Split Collection"
+      version="1.0.0"
+      tool_type="split_collection_by_number">
+    <description>by integer into multiple smaller collections</description>
+    <type class="SplitCollectionByNumberTool" module="galaxy.tools" />
+    <action module="galaxy.tools.actions.model_operations"
+            class="ModelOperationToolAction"/>
+    <inputs>
+        <param type="data_collection" name="input" label="Input Collection"
+            help="Collection to split into different parts"
+            optional="false"/>
+        <conditional name="how">
+            <param type="select" name="how_split" label="How should collections be created">
+                <option value="number_of_collections" selected="true">Number of collections wanted.</option>
+                <option value="maximum_elements">Maximum number of elements in each collection.</option>
+            </param>
+            <when value="number_of_collections">
+                <param type="integer" name="num_collection"
+                    value="" min="2"
+                    label="Number of collections"
+                    help="Split the large collection into this many collections (if possible)" />
+            </when>
+            <when value="maximum_elements">
+                <param type="integer" name="num_collection"
+                value="" min="1"
+                label="Maximum Number of Elements" 
+                help="Maximum number of elements to be found in each new collection" />
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <collection format_source="input" type_source="input" name="output" label="Split using $how.how_split with $how.num_collection " >
+            <discover_datasets pattern="__default__" directory="./" visible="false" />
+        </collection>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input">
+                <collection type="list">
+                    <element name="in1" value="simple_line.txt" />
+                    <element name="in2" value="simple_line_alternative.txt" />
+                </collection>
+            </param>
+            <param name="how_split" value="maximum_elements" />
+            <param name="num_collection" value="1" />
+            <output_collection name="output" type="list">
```

Comment:
This is where the tool test is failing (https://jenkins.galaxyproject.org/job/docker-framework-py3/157/testReport/junit/test.functional/test_toolbox_pytest/test_tool___SPLIT_COLLECTION_BY_NUMBER___test_1_/)
I think you may need to use `output_0`, `output_1` etc as parameter name here.

You can run the tool tests in isolation with `run_tests.sh -framework -id __SPLIT_COLLECTION_BY_NUMBER__ `